api/app_analytics/analytics_db_service.py

```diff
@@ -14,31 +14,103 @@
     FeatureEvaluationBucket,
     Resource,
 )
+from dateutil.relativedelta import relativedelta
 from django.conf import settings
 from django.db.models import Sum
 from django.utils import timezone
 
 from environments.models import Environment
 from features.models import Feature
+from organisations.models import Organisation
 
 ANALYTICS_READ_BUCKET_SIZE = 15
 
 
 def get_usage_data(
-    organisation, environment_id=None, project_id=None
-) -> List[UsageData]:
+    organisation: Organisation,
+    environment_id: int | None = None,
+    project_id: int | None = None,
+    period: str | None = None,
+) -> list[UsageData]:
+    now = timezone.now()
+
+    date_stop = date_start = None
+    period_starts_at = period_ends_at = None
+
+    if period == "current_billing_period":
+        if not getattr(organisation, "subscription_information_cache", None):
+            return []
+        sub_cache = organisation.subscription_information_cache
+        starts_at = sub_cache.current_billing_term_starts_at
+        month_delta = relativedelta(now, starts_at).months
+        period_starts_at = relativedelta(months=month_delta) + starts_at
+        period_ends_at = now
+        date_start = f"-{(now - period_starts_at).days}d"
+        date_stop = "now()"
+
+    elif period == "previous_billing_period":
+        if not getattr(organisation, "subscription_information_cache", None):
+            return []
+        sub_cache = organisation.subscription_information_cache
+        starts_at = sub_cache.current_billing_term_starts_at
+        month_delta = relativedelta(now, starts_at).months - 1
+        month_delta += relativedelta(now, starts_at).years * 12
+        period_starts_at = relativedelta(months=month_delta) + starts_at
+        period_ends_at = relativedelta(months=month_delta + 1) + starts_at
+        date_start = f"-{(now - period_starts_at).days}d"
+        date_stop = f"-{(now - period_ends_at).days}d"
+
+    elif period == "90_day_period":
+        period_starts_at = now - relativedelta(days=90)
+        period_ends_at = now
+        date_start = "-90d"
+        date_stop = "now()"
+
     if settings.USE_POSTGRES_FOR_ANALYTICS:
+        if period_starts_at:
+            assert period_ends_at
+            return get_usage_data_from_local_db(
+                organisation=organisation,
+                environment_id=environment_id,
+                project_id=project_id,
+                date_start=period_starts_at,
+                date_stop=period_ends_at,
+            )
         return get_usage_data_from_local_db(
-            organisation, environment_id=environment_id, project_id=project_id
+            organisation=organisation,
+            environment_id=environment_id,
+            project_id=project_id,
+        )
+
+    if date_start:
+        assert date_stop
```

Comment:
The assertions have to stay in the module because they are not common to the code that lives in where they are calling. It's asserting that we're understanding the world correctly, not assertions that we're calling the function correctly.

But I did simplify the code by using kwargs and keeping the number of function calls to a minimum 👍🏼